pilot/pkg/xds/proxy_dependencies.go

```diff
@@ -35,6 +35,13 @@ func ConfigAffectsProxy(req *model.PushRequest, proxy *model.Proxy) bool {
 		return true
 	}
 
+	// When endpoints for headless service change, we should push if EnableHeadlessService is true
```

Comment:
I am not sure I am following.. since this checks `Full && ...`, wouldn't we push it anyways? The only way we would filter is if it was filtered out by a Sidecar, but in that case we really don't want to push?

Reply:
Are you saying full pushes are always pushed? But when we pushConnection, do not we call this for every proxy?

Reply:
No, it isn't always, but ConfigAffectsProxy scopes based on Sidecar which seems fine...?

Reply:
I have changed it a bit. Hopefully that should address this part. PTAL and let me know.